src/aws/cloudwatch-sink.cpp

```diff
@@ -395,22 +403,29 @@ void CloudWatchSink::stop_impl()
     {
         return;
     }
-    is_running_ = false;
-    if (started_by_current_process() && cloudwatch_logs_thread_ && cloudwatch_logs_thread_->joinable())
+    try
     {
-        cloudwatch_logs_thread_->join();
+        is_running_ = false;
+        if (started_by_current_process() && cloudwatch_logs_thread_ && cloudwatch_logs_thread_->joinable())
+        {
+            cloudwatch_logs_thread_->join();
+        }
+        else if (!started_by_current_process())
+        {
+            // We are in a forked process, calling `reset()` will hang the client.
+            cloudwatch_logs_thread_.release();
+            aws_cloudwatch_client_.release();
+            logs_mtx_.fork_reset();
+            sequence_tokens_mtx_.fork_reset();
+        }
+        cloudwatch_logs_thread_.reset();
+        aws_cloudwatch_client_.reset();
+        logs_queue_.clear();
     }
-    else if (!started_by_current_process())
+    catch (const std::exception& e)
     {
-        // We are in a forked process, calling `reset()` will hang the client.
-        cloudwatch_logs_thread_.release();
-        aws_cloudwatch_client_.release();
-        logs_mtx_.fork_reset();
-        sequence_tokens_mtx_.fork_reset();
+        // Ignored, just so the thread itself will not die
```

Comment:
use the 'internal logging' for debug purposes?
also, in what scenario does these raise an exception? please add comment.

Reply:
those places are critical on forking, we do not want to print anything, to not cause a lock on fork issue, either from stdout or from our logging
and this is just a safety mechanism to not segfault mainly